roles/sideload_kernel/tasks/main.yml

```diff
@@ -0,0 +1,124 @@
+
+- name: Install community.kubernetes dependencies
+  block:
+    - name: Prefer package manager
+      become: true
+      ansible.builtin.package:
+        name: python3-kubernetes
+        state: present
+  rescue:
+    - name: Fallback to pip installation
+      ansible.builtin.pip:
+        name: kubernetes
+
+- name: Sideload-kernel job namespace
+  community.kubernetes.k8s:
+    state: present
+    api_version: v1
+    kind: Namespace
+    name: "{{ sideload_kernel_namespace }}"
+  environment: "{{ k8s_auth }}"
+
+- name: Sideload-kernel configmap
+  community.kubernetes.k8s:
+    state: present
+    apply: true
+    definition:
+      apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        name: sideload-kernel
+        namespace: "{{ sideload_kernel_namespace }}"
+      data:
+        KERNEL_URI: "{{ sideload_kernel_uri }}"
+        flip_kernel: "{{ lookup('file', 'flip_kernel') }}"
+  environment: "{{ k8s_auth }}"
+  register: configmap_created
+
+- name: Clean out any previous jobs
+  community.kubernetes.k8s:
+    state: absent
+    api_version: batch/v1
+    kind: Job
+    name: flip-kernel
+    namespace: "{{ sideload_kernel_namespace }}"
+  environment: "{{ k8s_auth }}"
+  when: configmap_created.changed or sideload_kernel_force | bool
+
+- name: Sideload-kernel job
+  community.kubernetes.k8s:
+    state: present
+    definition:
+      apiVersion: batch/v1
+      kind: Job
+      metadata:
+        name: flip-kernel
+        namespace: "{{ sideload_kernel_namespace }}"
+      spec:
+        backoffLimit: 3
+        # TODO: Add a selector for multi-node cluster cases?
+        template:
+          spec:
+            containers:
+              - command:
+                  - /bin/bash
+                args:
+                  - "-c"
+                  - "install /script/flip_kernel /host/tmp && chroot /host /tmp/flip_kernel ${KERNEL_URI}"
+                env:
+                  - name: KERNEL_URI
+                    valueFrom:
+                      configMapKeyRef:
+                        name: sideload-kernel
+                        key: KERNEL_URI
+                image: "{{ sideload_kernel_base_image }}"
+                imagePullPolicy: IfNotPresent
+                name: flipper
+                securityContext:
+                  privileged: true
+                  runAsUser: 0
+                volumeMounts:
+                  - mountPath: /host
+                    name: host
+                  - mountPath: /script
+                    name: script
+            hostIPC: true
+            hostNetwork: true
+            hostPID: true
+            restartPolicy: Never
+            volumes:
+              - name: host
+                hostPath:
+                  path: /
+                  type: Directory
+              - name: script
+                configMap:
+                  name: sideload-kernel
+  environment: "{{ k8s_auth }}"
+
+- ansible.builtin.debug: # noqa name[missing]
```

Comment:
Missing task name?

Reply:
Intentional.  I've never seen the point in having names for debug tasks, since they just print a thing that's more useful than any static name I could come up with for the task.

That said, I wouldn't mind naming it, if that's an important convention.